### PR TITLE
docs: update api

### PR DESCRIPTION
**Pull request by cbt1**
Attempts to fix a few errors in the spec.

I should say there are still plenty of things that are wrong or inconsistent. I'm hoping to fix most of those in smaller (easier to review) PR:s.

### CODE REVIEW

File: packages/picasso.js/src/core/chart-components/labels/labels.js [hunk 1]
```diff
@@ -2,15 +2,20 @@ import strategies from './strategies';
 
 /**
  * @typedef {object} ComponentLabels
- * @property {string} [type='labels']
+ * @extends ComponentSettings
  */
 
 /**
+ * Component settings
  * @typedef {object} ComponentLabels.settings
```
Comment (T-Wizard) [inline packages/picasso.js/src/core/chart-components/labels/labels.js:10]:
```suggestion
 * @typedef {object} ComponentLabels~Settings
```
to be more consistent with the changes on other components

File: packages/picasso.js/src/core/chart-components/labels/labels.js [hunk 2]
```diff
@@ -2,15 +2,20 @@ import strategies from './strategies';
 
 /**
  * @typedef {object} ComponentLabels
- * @property {string} [type='labels']
+ * @extends ComponentSettings
  */
```
Comment (T-Wizard) [inline packages/picasso.js/src/core/chart-components/labels/labels.js:6]: maybe an example?
(to specify the value type should have)

File: packages/picasso.js/src/core/chart-components/legend-seq/index.js [hunk 3]
```diff
@@ -1,5 +1,11 @@
 import legendSeq from './legend-seq';
 
+/**
+ * @typedef {object} ComponentLegendSeq
+ * @extends ComponentSettings
+ * @property {string} scale Reference to sequential color scale
+ */
```
Comment (T-Wizard) [inline packages/picasso.js/src/core/chart-components/legend-seq/index.js:7]: missing?
@property {ComponentLegendSeq.setting} settings Legend settings
+
example?

File: packages/picasso.js/src/core/chart-components/labels/labels.js (hunk 1 above)
Comment (cbt1) [reply packages/picasso.js/src/core/chart-components/labels/labels.js:10]: Yeah so I would like to use `~` only when we have multiple different settings objects. Like in the case of the axis component. If there is only a single setting it should use `.` IMO. I'll update it.

File: packages/picasso.js/src/core/chart-components/labels/labels.js (hunk 2 above)
Comment (cbt1) [reply packages/picasso.js/src/core/chart-components/labels/labels.js:6]: All components should have examples (actually most definitions should IMO). I'm just not gonna have the time to fix that in this PR.

File: packages/picasso.js/src/core/chart-components/legend-seq/index.js (hunk 3 above)
Comment (cbt1) [reply packages/picasso.js/src/core/chart-components/legend-seq/index.js:7]: `ComponentLegendSeq.settings` is definied in `legend-seq.js`?

File: packages/picasso.js/src/core/chart-components/labels/labels.js (hunk 1 above)
Comment (cbt1) [reply packages/picasso.js/src/core/chart-components/labels/labels.js:10]: I've updated the spec to use `.` when possible and `~` only for special cases

File: packages/picasso.js/src/core/chart-components/legend-seq/index.js (hunk 3 above)
Comment (T-Wizard) [reply packages/picasso.js/src/core/chart-components/legend-seq/index.js:7]: the type `ComponentLegendSeq.settings` is definied in `legend-seq.js`
but I can not find that it defines that an object of type ComponentLegendSeq has a `settings` property (of type `ComponentLegendSeq.settings`). but maybe it is there and I could just not find it.

**Comment by T-Wizard** [comment]
`label` in `ComponentPoint.settings` looks like should be documented but is not
(it that intentional or you did you just miss it?)

File: packages/picasso.js/src/core/chart-components/property-resolver/index.js [hunk 4]
```diff
@@ -18,26 +18,44 @@ function isPrimitive(x) {
...
+ * {
+ *  scale: '<name-of-scale>',
+ *  ref: '<data-property>'
+ * }
+ *
+ * // or explictly resolve the datum using callback function
+ * {
+ *  scale: '<name-of-scale>',
+ *  fn: (d) => Math.min(0, d.value);
+ * }
```
Comment (T-Wizard) [inline packages/picasso.js/src/core/chart-components/property-resolver/index.js:44]: consider removing `scale: '<name-of-scale>',`
I don't think is relevant for this example.

and I think `fn: (d) => Math.min(0, d.datum.x.value);` is a more realistic function

**Review by T-Wizard — APPROVED** [review]

File: packages/picasso.js/src/core/chart-components/legend-seq/index.js (hunk 3 above)
Comment (cbt1) [reply packages/picasso.js/src/core/chart-components/legend-seq/index.js:7]: Yes so I believe you don't have to define the property as well

File: packages/picasso.js/src/core/chart-components/property-resolver/index.js (hunk 4 above)
Comment (cbt1) [reply packages/picasso.js/src/core/chart-components/property-resolver/index.js]: Fixed

**Pull request merged**